### PR TITLE
picard: build 32bit package

### PR DESCRIPTION
**Pull request by phw**
This is an attempt to make picard available as a 32 bit package.

@Begasus Can you take a look? To be honest I cannot fully wrap my head around this secondary architecture thing and I cannot find any documentation about it. I just copied this from other recipes.

### CODE REVIEW

File: media-sound/picard/picard-2.1.3.recipe [hunk 1]
```diff
@@ -4,35 +4,36 @@ HOMEPAGE="https://picard.musicbrainz.org/"
...
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	picard = $portVersion
+	picard${secondaryArchSuffix} = $portVersion
 	cmd:picard = $portVersion
 	"
 REQUIRES="
-	haiku
+	haiku${secondaryArchSuffix}
```
Comment (Begasus) [inline media-sound/picard/picard-2.1.3.recipe]: Remove the brackets:
```diff
-	haiku${secondaryArchSuffix}
+	haiku$secondaryArchSuffix
```

File: media-sound/picard/picard-2.1.3.recipe [hunk 2]
```diff
@@ -4,35 +4,36 @@ HOMEPAGE="https://picard.musicbrainz.org/"
...
 	cmd:python3.6
-	pyqt_python3
+	pyqt_python3${secondaryArchSuffix}
```
Comment (Begasus) [inline media-sound/picard/picard-2.1.3.recipe]: just pyqt_python3 will do (no need for secondaryArchSuffix)
PS we only use the brackets for a few things :)

File: media-sound/picard/picard-2.1.3.recipe [hunk 3]
```diff
@@ -4,35 +4,36 @@ HOMEPAGE="https://picard.musicbrainz.org/"
...
 
 BUILD_REQUIRES="
-	haiku_devel
+	haiku${secondaryArchSuffix}_devel
 	cmd:gcc
 	cmd:msgfmt
 	cmd:python3.6
 	setuptools_python36
-	pyqt_python3
+	pyqt_python3${secondaryArchSuffix}
```
Comment (Begasus) [inline media-sound/picard/picard-2.1.3.recipe]: Same here, without the secondaryArchSuffix

**Comment by Begasus** [comment]
FWIW I pushed my changes to build it on gcc2h to https://github.com/Begasus/haikuports/commit/70a8099f9561fe94fe584281912498823a3080f5

**Comment by Begasus** [comment]
Also left a comment on the chromaprint recipe https://github.com/haikuports/haikuports/pull/4007/commits/1c1a4d93b17b849f40029a1accd6727d1f6cf599#r317354235

**Comment by phw** [comment]
@Begasus Thanks for this, I replaced my commit with yours for this PR. Thanks for your clarification on the architectures also. What do the prefixes `!` and `?` in `!x86_gcc2 ?x86` do? Is `!` "don't build" or priority?

**Comment by phw** [comment]
Also tested on x86_64, still works ;)

File: media-sound/picard/picard-2.1.3.recipe [hunk 4]
```diff
@@ -4,35 +4,38 @@ HOMEPAGE="https://picard.musicbrainz.org/"
...
-	haiku_devel
-	cmd:gcc
-	cmd:msgfmt
-	cmd:python3.6
+	haiku${secondaryArchSuffix}_devel
 	setuptools_python36
-	pyqt_python3
+	"
+BUILD_PREREQUIRES="
+	cmd:fpcalc$secondaryArchSuffix
```
Comment (phw) [inline media-sound/picard/picard-2.1.3.recipe]: @Begasus I don't think we need this for build, or was there a specific reason you added it?

Comment (Begasus) [reply media-sound/picard/picard-2.1.3.recipe]: It was already in there (but as a left the comment on chromaprint, cmd:fpcalc isn't provided in the recipe, but the binary is in $prefix/bin so cmd:fpcalc is correct), for now I used this to bypass the problem, haven't checked if it's really needed

**Comment by Begasus** [comment]
! tells haikuporter the build is broken for that architecture ? tells haikuported the build hasn't been tested (for ?x86 it's ok, we don't have a release for x86 haiku version (gcc8 only))

**Comment by phw** [comment]
> ! tells haikuporter the build is broken for that architecture ?

See how this is confusing me to no end :D So we get a build for gcc2h by telling the system the build for x86_gcc2 is broken? Why?

**Comment by Begasus** [comment]
> > ! tells haikuporter the build is broken for that architecture ?
> 
> See how this is confusing me to no end :D So we get a build for gcc2h by telling the system the build for x86_gcc2 is broken? Why?

Because x86_gcc2 is the primary target (using gcc2) which can't build Picard :) hence it's broken "!"

File: media-sound/picard/picard-2.1.3.recipe (hunk 4 above)
Comment (Begasus) [reply media-sound/picard/picard-2.1.3.recipe]: @phw correct not needed so can be removed in the recipe

**Comment by Begasus** [comment]
LGTM only remove the requirement for fpcalc :) (after you tested on x86_64) :)

**Comment by phw** [comment]
> LGTM only remove the requirement for fpcalc :) (after you tested on x86_64) :)

fpcalc removed from `BUILD_PREREQUIRES` and build tested on x86_64. Ready for merging then.

**Pull request merged**